test/gsl-lite.t.hpp

```diff
@@ -31,7 +31,9 @@
 #ifdef __clang__
 # pragma clang diagnostic ignored "-Wstring-conversion"
 # pragma clang diagnostic ignored "-Wunused-parameter"
+#ifndef __APPLE_CC__
```

Comment:
What exactly is the issue here? Does Apple's flavor of Clang not support `-Wunused-template`?